### PR TITLE
[nrf noup] Update Mbed TLS config file defines

### PR DESCRIPTION
**Pull request by vili-nordic**
MBEDTLS_USER_CONFIG_FILE has been replaced with MBEDTLS_PSA_CRYPTO_CONFIG_FILE and MBEDTLS_PSA_CRYPTO_USER_CONFIG_FILE. This commit updates Matter's CMakeLists to reflect this.

Needed for upmerge https://github.com/nrfconnect/sdk-nrf/pull/13662

### CODE REVIEW

**Review by SebastianBoe — APPROVED** [review]

File: config/nrfconnect/chip-module/Kconfig.defaults [hunk 1]
```diff
@@ -337,6 +337,9 @@ config MBEDTLS_SHA256_C
 config MBEDTLS_PK_C
     default y
 
+config MBEDTLS_PKCS5_C
+    default y
```
Comment (LuDuda) [inline config/nrfconnect/chip-module/Kconfig.defaults:341]: Why this is needed?

File: config/nrfconnect/chip-module/Kconfig.defaults [hunk 2]
```diff
@@ -337,6 +337,9 @@ config MBEDTLS_SHA256_C
 config MBEDTLS_PK_C
     default y
 
+config MBEDTLS_PKCS5_C
```
Comment (LuDuda) [inline config/nrfconnect/chip-module/Kconfig.defaults:340]: I don't see any `MBEDTLS_PKCS5_C` usage in the code. Why this is needed?

Comment (LuDuda) [reply config/nrfconnect/chip-module/Kconfig.defaults:340]: Oh i found it!
https://github.com/nrfconnect/sdk-connectedhomeip/blob/master/src/crypto/CHIPCryptoPALmbedTLS.cpp#L358

**Review by LuDuda — APPROVED** [review]
Approving, but we would like to run nightly tests on manifest PR today.

**Review by Vge0rge — APPROVED** [review]

**Comment by LuDuda** [comment]
> Approving, but we would like to run nightly tests on manifest PR today.

Btw. We have not found regression (in functionality) 👍

**Pull request merged**